test/e2e/testsuites/podfailovertest.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"context"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	scale "k8s.io/api/autoscaling/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	v1alpha1ClientSet "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/client/clientset/versioned/typed/azuredisk/v1alpha1"
+	"sigs.k8s.io/azuredisk-csi-driver/test/e2e/driver"
+)
+
+//  will provision required PV(s), PVC(s) and Pod(s)
+// Pod should successfully be re-scheduled on failover in a cluster with AzDriverNode and AzVolumeAttachment resources
+type PodFailover struct {
+	client                 clientset.Interface
+	CSIDriver              driver.DynamicPVTestDriver
+	AzDiskClientSet        v1alpha1ClientSet.DiskV1alpha1Interface
+	AzNamespace            string
+	Pod                    PodDetails
+	Volume                 VolumeDetails
+	StorageClassParameters map[string]string
+}
+
+func (t *PodFailover) Run(client clientset.Interface, namespace *v1.Namespace, schedulerName string) {
```

Comment:
Since a similar test already exists in the scheduler extender test suite, I would recommend adding the missing node cardon to [that test ](https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/5422820c62b9160f9e94e846b4903d258981ef34/test/e2e/testsuites/azdiskschedulerextender_pod_scheduling_on_failover.go#L44) and for a this failover test I think it would be nice to make it a stress test instead. We could create more pods and fail them on multiple thread and measure the time it takes to reschedule them with and without the scheduler extender. This would be somewhat similar to [the stress unittest](https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/5422820c62b9160f9e94e846b4903d258981ef34/pkg/azdiskschedulerextender/azdiskschedulerextender_test.go#L558) in the scheduler extender but would measure the performance in a real cluster.

Reply:
Agree that having this test as a part of dynamic provisioning tests makes sense and eliminates some of the of the duplication in scheduler extender tests. Since dynamic provisioning tests will run both for default scheduler and scheduler extender we would eventually delete the failover test from scheduler extender test.